jest.config.js

```diff
@@ -52,10 +52,20 @@ module.exports = {
     // Use babel-jest to transpile tests with the next/babel preset
     // https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object
     "^.+\\.(js|jsx|ts|tsx)$": [
-      "babel-jest",
+      "@swc/jest",
       {
-        presets: ["next/babel"],
```

Comment:
Eh, so it seems that Next is vendoring all Babel transforms and even the canary version of Next is still using old ones that do not support `satisfies`. It's virtually impossible for you to work around this on your own - unless you'd stop using the `next/babel` and just "inline" what they are using there but this time referring to the "real" packages.

I opened a PR to fix this: https://github.com/vercel/next.js/pull/51962

Reply:
Note that switching to `@swc/jest` is the right call to do at some point anyway so this is as good time as ever :P